src/IO/WriteBufferFromS3.cpp

```diff
@@ -164,34 +170,82 @@ void WriteBufferFromS3::writePart()
         LOG_WARNING(log, "Maximum part number in S3 protocol has reached (too many parts). Server may not accept this whole upload.");
     }
 
+    auto part_data = std::make_shared<Aws::StringStream>();
+    temporary_buffer.swap(part_data);
+
+    size_t part_number = part_tags.size() + 1;
+    auto part_tag = std::make_shared<String>();
+    part_tags.emplace_back(part_tag);
+    auto thread_group = CurrentThread::getGroup();
+    auto write_job = [&, part_number, part_tag, part_data, thread_group]
+    {
+        doWritePart(part_data, part_number, part_tag, thread_group);
+    };
+
+    if (writing_thread_pool)
+        writing_thread_pool->scheduleOrThrowOnError(std::move(write_job));
+    else
+        write_job();
```

Comment:
Better to always use thread pool with 4 threads by default.

Reply:
Even if multipart_write_thread_pool_size=0? I'm voting for not do any thread pool in this case, or use multipart_write_thread_pool_size=1 by default.

Reply:
It is 1 by default now. In cases when we have <1 uploads at a time, it may be useful to set greater value.